### PR TITLE
Add com.github.torikulhabib.niki

### PR DESCRIPTION
**Pull request by torikulhabib**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. [**Link:**](https://github.com/torikulhabib/niki)
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by torikulhabib** [comment]
bot, build com.github.torikulhabib.niki

File: com.github.torikulhabib.niki.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+app-id: com.github.torikulhabib.niki
+runtime: org.gnome.Platform
+runtime-version: '41'
+sdk: org.gnome.Sdk
+command: com.github.torikulhabib.niki
+finish-args:
+  - '--share=network'
+  - '--socket=session-bus'
```
Comment (hfiguiere) [inline com.github.torikulhabib.niki.yml:8]: You need to figure what dbus name are need. Addind the `session-bus` is usually not an option.

File: com.github.torikulhabib.niki.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  - '--share=ipc'
+  - '--socket=fallback-x11'
+  - '--socket=wayland'
+  - '--socket=pulseaudio'
+  - '--socket=x11'
```
Comment (hfiguiere) [inline com.github.torikulhabib.niki.yml:13]:
```suggestion
```

File: com.github.torikulhabib.niki.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      - type: git
+        url: https://github.com/GNOME/gupnp.git
+        tag: gupnp-1.2.4
+  - name: libgupnp-av-1.0-2
+    buildsystem: meson
+    sources:
+      - type: git
+        url: https://github.com/GNOME/gupnp-av.git
+        tag: gupnp-av-0.12.11
+  - name: libcanberra
```
Comment (hfiguiere) [inline com.github.torikulhabib.niki.yml:39]: please use the manifest `shared-modules` as it contain important patches.

File: com.github.torikulhabib.niki.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    buildsystem: autotools
+    sources:
+      - type: archive
+        url: http://0pointer.de/lennart/projects/libcanberra/libcanberra-0.30.tar.xz
+        sha256: c2b671e67e0c288a69fc33dc1b6f1b534d07882c2aceed37004bf48c601afa72
+  - name: taglib
+    buildsystem: cmake
```
Comment (hfiguiere) [inline com.github.torikulhabib.niki.yml:46]:
```suggestion
    buildsystem: cmake-ninja
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: com.github.torikulhabib.niki.yml (hunk 3 above)
Comment (hfiguiere) [reply com.github.torikulhabib.niki.yml:38]: I was talking about libcanberra.

Comment (hfiguiere) [reply com.github.torikulhabib.niki.yml]: still not what I was saying.

**Comment by hfiguiere** [comment]
bot, build com.github.torikulhabib.niki

File: com.github.torikulhabib.niki.yml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+      - type: git
+        url: https://github.com/GNOME/gupnp.git
+        tag: gupnp-1.2.4
+  - name: libgupnp-av-1.0-2
+    buildsystem: meson
+    sources:
+      - type: git
+        url: https://github.com/GNOME/gupnp-av.git
+        tag: gupnp-av-0.12.11
+  - name: libcanberra-0.30
```
Comment (nedrichards) [inline com.github.torikulhabib.niki.yml:38]: this is in the shared-modules https://github.com/flathub/shared-modules/ you should pull it in from there please.

**Comment by nedrichards** [comment]
While browsing I get a fair few

> Error: Your GStreamer installation is missing a plug-in.

issues - you may need to bundle a wider set of plugins.

**Comment by nedrichards** [comment]
For more clarity on what I mean about codecs see the kind of setup that Totem has to be gstreamer based and support 'everything' https://github.com/flathub/org.gnome.Totem.Devel/blob/beta/org.gnome.Totem.json

**Comment by torikulhabib** [comment]
bot, build com.github.torikulhabib.niki

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
+### Please confirm your submission meets all the criteria
```
Comment (hfiguiere) [inline .github/pull_request_template.md:1]: this shouldn't be in there.

File: .github/workflows/stale.yml [hunk 7]
```diff
@@ -0,0 +1,15 @@
+name: "Close stale PR"
```
Comment (hfiguiere) [inline .github/workflows/stale.yml:1]: nor the workflow.

File: com.github.torikulhabib.niki.yml [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+sdk: org.gnome.Sdk
+command: com.github.torikulhabib.niki
+finish-args:
+  - '--share=network'
+  - '--talk-name=org.freedesktop.thumbnails.Thumbnailer1'
+  - '--share=ipc'
+  - '--socket=fallback-x11'
+  - '--socket=wayland'
+  - '--socket=pulseaudio'
+  - '--socket=x11'
```
Comment (hfiguiere) [inline com.github.torikulhabib.niki.yml:13]:
```suggestion
  - '--socket=x11'
```

File: com.github.torikulhabib.niki.yml [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+        url: https://github.com/GNOME/gupnp.git
+        tag: gupnp-1.2.4
+  - name: libgupnp-av-1.0-2
+    buildsystem: meson
+    sources:
+      - type: git
+        url: https://github.com/GNOME/gupnp-av.git
+        tag: gupnp-av-0.12.11
+  - name: taglib
+    buildsystem: cmake
```
Comment (hfiguiere) [inline com.github.torikulhabib.niki.yml:53]:
```suggestion
    buildsystem: cmake-ninja
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: com.github.torikulhabib.niki.yml [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+  - 'shared-modules/libcanberra/libcanberra.json'
+  - 'codecs/liba52.json'
+  - 'codecs/libmpeg2.json'
+  - 'codecs/libdvdcss.json'
+  - 'codecs/libdvdread.json'
+  - 'codecs/libdvdnav.json'
+  - 'codecs/libass.json'
+  - 'codecs/ffmpeg.json'
+  - 'codecs/gstreamer.json'
+  - 'codecs/pipewire.json'
```
Comment (hfiguiere) [inline com.github.torikulhabib.niki.yml:32]: these manifests are missing from the repository too.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by nedrichards** [comment]
bot, build com.github.torikulhabib.niki

**Comment by hfiguiere** [comment]
build fail because of what I mentionned in my review comments.

**Comment by nedrichards** [comment]
Yup, I know - sometimes a failure makes it clearer that those comments are there for a reason!

**Pull request closed without merging**